Userland/Libraries/LibJS/Runtime/SafeStringBuilder.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/SafeStringBuilder.h>
+
+namespace JS {
+
+SafeStringBuilder::SafeStringBuilder(VM& vm)
+    : m_vm(vm)
+{
+}
+
+ThrowCompletionOr<void> SafeStringBuilder::append(char ch)
+{
+    if (try_append(ch).is_error())
+        return m_vm.throw_completion<RangeError>(ErrorType::NotEnoughMemoryToAllocate, length() + 1);
```

Comment:
Please use `InternalError` for these, I'm not convinced that we should be using `RangeError` for unspec'd OOM situations. Imagine someone doing something that catches such an OOM RangeError, assuming it's related to an ArrayBuffer allocation failure and continuing even though it's totally unrelated string allocation failure.